### ISSUE TITLE
Interesting data to visualize

### ISSUE THREAD

**Issue by NigelKibodeaux** [codeforanchorage/bus-request-loggeR#1]
Hans, this is great! Here are a couple of things that _I'd_ be interested in seeing:
- Stacked bar graph of total requests each day, with each bar consisting of lat/lon, Twilio with stop number, Twilio with address, web with stop number, web with address
- Some kind of map visualization of which stops are requested. There is a file in the other repo that associates stop numbers with a lat/lon

**Comment by brendanbabb** [comment]
We should also consider viewing requests like "Help" or "How does this work" Maybe not in the stacked bar graph but to get an idea if people are having questions.

If we can I think we should have a lat/lon with every request in the log. Then we could also run an easy sped up animation of where requests were coming from on a map over a day or week.

**Comment by NigelKibodeaux** [comment]
@brendanbabb The requests that came in with lat/lon are differentiated in the log by having that. I think we should look up lat/lon of the returned stop for mapping purposes to avoid mixing output coordinates with input coordinates.

**Comment by brendanbabb** [comment]
@NigelKibodeaux Yes, that is what I want, the resulting lat longs, not the requests.

**Comment by NigelKibodeaux** [comment]
In response to Brendan's first paragraph, maybe we should group requests that couldn't be fulfilled in the chart. These are the ones that have an "input" but no "stop".

**Comment by thermokarst** [comment]
This is really cool, but I think we need to make sure that we include appropriate data collection disclaimers/waivers on the bus app before we start sharing these data.

**Comment by hansthompson** [comment]
Thanks for taking a look guys! I added an issue as Visual Requests. 

In regards to disclaimers for the main site, what is most appropriate?  A message on the site that says "Requests through this service are logged for statistical analysis"? Is there a standard way to approach this with text messaging?  

My opinion is that users typically need to agree to making their location already known if they are using a common web browser though a pop up or changing the default settings.  I would suggest maybe we could encrypt the json log, share the key privately between us and put the file in the .gitignore of this log analysis repo. Thoughts?

**Comment by NigelKibodeaux** [comment]
We could not store the user's precise location, just that it was a GPS
request.  We'll have the nearest bus stop.

**Comment by hansthompson** [comment]
Or we could just go to three or four significant figures in the log. That would only approximate the location.

**Comment by thermokarst** [comment]
I don't think the specifics matter much (e.g. encryption, significant digits, etc.), i just brought it up as an aside. If we are collecting _any_ data, we need to tell people that.

**Comment by NigelKibodeaux** [comment]
I don't think this is possible with the SMS interface but we could have a
"privacy policy" link on the web. If we stop storing the GPS lat/lon, we're
not collecting any personally identifiable data though. To me, it's not an
issue.

**Comment by ghumphr** [comment]
It's not a bad idea to put a notice somewhere stating that under no
circumstances will you sell the client's data to any third party.

**Comment by thermokarst** [comment]
Sorry, I wasn't trying to make a mountain out of a molehill.

@NigelKibodeaux: I don't think we need to worry about not storing GPS location, in fact, I think it makes Hans' loggeR even more interesting. This was just a high-level note --- we need to be good citizens. Its one thing if we were collecting the information internally (like for application logging only), but this loggeR project turns these data into a product (a transparent, open product, but still a product). This has nothing to do with personally identifiable data, this has to do with transparency. All we need is a little "Info" or "About" somewhere on the page that says "We track [anonymous?] usage statistics/metrics, blah blah."

@ghumphr: I think that is besides the point --- we need to ensure that somewhere, somehow, people know that _we_ are collecting usage metrics, and potentially utilizing or publishing the information. If we want to tell them that we aren't selling their info, thats fine, but the point is we need to be transparent about the collection process in general.

**Comment by ghumphr** [comment]
There are standard notices used for that sort of thing that usually do
discuss purposes to which the data might be used. I assume you aren't
publishing individual-level information... publishing summary information
would be ok, and it's not a bad idea to post a notice stating that you may
do so.  I don't have one sitting in front of me, but I am sure we can draft
one.

**Comment by ghumphr** [comment]
How about: "We respect your privacy and will not share your personal information to any third party except for bona fide law enforcement purposes such as warrant service or bail bond recovery."

**Comment by thermokarst** [comment]
@ghumphr: Good start, but I think we should leave out any specific examples, including law enforcement (if we are concerned about that, then we need to consult with a lawyer --- I don't think that is the case though) and bail bond recovery (also: under what circumstances would a bail bondsman hit this app's logs up???)

How about something much more generic:

> NOTICE: Usage metrics and search queries are collected for data interpretation purposes. Analyses may be found here: [URL to loggeR app when it is up and running].

I just want to make sure that people that use the app aren't accidentally disclosing stuff in their searches (like their home address or name/phone #), and then holding us accountable.
